### PR TITLE
feat(styles): Add darkreader dark theme option

### PR DESCRIPTION
**Pull request by Anemy**
Extension of https://github.com/mongodb-js/compass/pull/2444 and https://github.com/mongodb-js/compass/pull/1960

This enables using [darkreader](https://github.com/darkreader/darkreader) to enable darkmode and listens to native ui color preference for darkmode. Before merging we're going to look into updating the sidebar colors and sync with our design team.

This is a temporary solution while we rework all of Compass' style to work with a dark theme. The theme can be chosen in the menu using the radio options (OS Default, Dark, Light).

**Theme choosing options have been moved into a submenu of `Theme` in the general app menus since recording this video**

https://user-images.githubusercontent.com/1791149/143979355-06f42110-2fee-479e-ab43-8658689ea4d7.mp4

### CODE REVIEW

File: packages/compass/src/main/menu.ts [hunk 1]
```diff
@@ -55,36 +82,36 @@ function networkOptInDialogItem() {
...
       {
         label: 'Hide Others',
         accelerator: 'Command+Shift+H',
-        selector: 'hideOtherApplications:',
+        role: 'hideOthers',
       },
       {
         label: 'Show All',
-        selector: 'unhideAllApplications:',
+        role: 'unhide',
```
Comment (Anemy) [inline packages/compass/src/main/menu.ts:106]: Might want to move these fixes into a separate pr with a ticket attached
https://www.electronjs.org/docs/latest/api/menu-item#roles

**Comment by lerouxb** [comment]
Use OS theme should probably not say (Preview) as well?

**Review by lerouxb — APPROVED** [review]
👏

File: packages/compass/src/main/menu.spec.ts [hunk 2]
```diff
@@ -127,6 +127,24 @@ describe('CompassMenu', function () {
           {
             type: 'separator',
           },
+          {
+            checked: true,
+            label: 'Use OS Theme (Preview)',
```
Comment (Anemy) [inline packages/compass/src/main/menu.spec.ts:132]: I'm thinking let's move these options into the Compass menu -> preferences -> color theme. Instead of where they are currently in view. Will update

File: packages/compass/src/main/menu.ts [hunk 3]
```diff
@@ -55,7 +82,71 @@ function networkOptInDialogItem(): MenuItemConstructorOptions {
   };
 }
 
-function darwinCompassSubMenu(): MenuItemConstructorOptions {
+function trackThemeChanged(
+  newTheme: THEMES
+) {
+  CompassTelemetry.track({
+    event: 'Theme Changed',
```
Comment (Anemy) [inline packages/compass/src/main/menu.ts:89]: Added telemetry tracking
COMPASS-5329

**Comment by brendonlamb** [comment]
all i want for Christmas is my dark mode theme

thank you for your work on this! hopefully it's fully integrated/released soon :D
